pkg/scheduler/framework/interface.go

```diff
@@ -196,6 +196,9 @@ func (s *Status) Equal(x *Status) bool {
 	if s.code == Error {
 		return cmp.Equal(s.err, x.err, cmpopts.EquateErrors())
 	}
+	if s.failedPlugin != "" || x.failedPlugin != "" {
+		return s.failedPlugin == x.failedPlugin
+	}
```

Comment:
changes looks good to me, @Huang-Wei do you aware of any negative impact of this change?

Reply:
No need for the if statement

Reply:
well..I found an explanation... the [findNodesThatFitPod](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/generic_scheduler_test.go#L1065) function here will run to [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/generic_scheduler.go#L310), therefore, there will be no inconsistency between Status.failedPlugin and UnschedulablePlugins. If you manually modify all these fields like:

```
expected := framework.Diagnosis{
		NodeToStatusMap: framework.NodeToStatusMap{
			"1": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
			"2": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
			"3": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
		},
		UnschedulablePlugins: sets.NewString("AnotherFilter"),
	}
```

**cmp.Diff** is possible, and do not need to modify Equal.

@chendave

Reply:
the code in [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/generic_scheduler.go#L310)
so there will be no inconsistency between Status.failedPlugin and UnschedulablePlugins. If you manually modify all these fields like:

```
expected := framework.Diagnosis{
		NodeToStatusMap: framework.NodeToStatusMap{
			"1": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
			"2": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
			"3": framework.NewStatus(framework.Unschedulable, st.ErrReasonFake).WithFailedPlugin("AnotherFilter"),
		},
		UnschedulablePlugins: sets.NewString("AnotherFilter"),
	}
```

**cmp.Diff** is possible, and do not need to modify Equal.

@chendave

Reply:
> changes looks good to me, @Huang-Wei do you aware of any negative impact of this change?

when I reviewed this PR, this code wasn't there... And the latest code also removed it.

Reply:
>so there will be no inconsistency between Status.failedPlugin and UnschedulablePlugins. If you manually modify all these fields like:

Not follow here, shouldn't `cmp.Diff` expect to compare each of the fields here? Suppose `diagnosis` return struct where the  field plugin is `MatchFilter`,  and `expected` is defined as `FakeFilter` but the test pass, does that make sense?

Reply:
@Huang-Wei , code is here, not sure why the code is reverted.

https://github.com/kubernetes/kubernetes/blob/e326c00bc5c1fc1a9897793e97b1055837979c0c/pkg/scheduler/framework/interface.go#L214-L225

Reply:
Sorry I modified the code without explanation. I'm not sure that it is necessary to verify the failedPlugin, because from the code I said above, it will be locked when the failedPlugin is modified, and the result of manual modification may never appear. This part of the check is redundant.And..here is the code：
```
func (s *Status) Equal(x *Status) bool { 
  if s == nil || x == nil { 
   return s.IsSuccess() && x.IsSuccess() 
  } 
  if s.code != x.code { 
   return false 
  } 
  if s.code == Error { 
   return cmp.Equal(s.err, x.err, cmpopts.EquateErrors()) 
  } 
  if s.failedPlugin != "" || x.failedPlugin != "" {
  return s.failedPlugin == x.failedPlugin
  }
  return cmp.Equal(s.reasons, x.reasons) 
 } 
```
@Huang-Wei